### PR TITLE
drivers: can: Kconfig.rcar: Make options depend on its driver

### PR DESCRIPTION
**Pull request by yashi**
Need this so that the option won't pop out without R-Car selected.

Signed-off-by: Yasushi SHOJI <yashi@spacecubics.com>

### CODE REVIEW

**Review by galak — APPROVED** [review]

**Review by julien-massot — APPROVED** [review]
Thanks !

File: drivers/can/Kconfig.rcar [hunk 1]
```diff
@@ -11,6 +11,7 @@ config CAN_RCAR
```
Comment (julien-massot) [inline drivers/can/Kconfig.rcar:11]: Can we do with a if statement, it looks more logical to me, and will be coherent with mcp2515 driver ?

```suggestion
if CAN_RCAR

```

File: drivers/can/Kconfig.rcar [hunk 2]
```diff
@@ -11,6 +11,7 @@ config CAN_RCAR
 
 config CAN_RCAR_MAX_FILTER
 	int "Maximum number of concurrent active filters"
+	depends on CAN_RCAR
 	default 5
 	range 1 32
 	help
```
Comment (julien-massot) [inline drivers/can/Kconfig.rcar:17]:
```suggestion
	help
	bla bla
endif # CAN_MCP2515
```

File: drivers/can/Kconfig.rcar (hunk 1 above)
Comment (yashi) [reply drivers/can/Kconfig.rcar:11]: Sure.  But I thought we use `depends on` for single symbol.  (aka. https://github.com/zephyrproject-rtos/zephyr/commit/eddd98f8118709aa29c0f14785749199be224a41 and many other commits related to it)

Comment (julien-massot) [reply drivers/can/Kconfig.rcar:11]: indeed miss that one. So yes better use depends on

**Review by julien-massot — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by alexanderwachter — APPROVED** [review]

**Pull request merged**